matrix_client/client.py

```diff
@@ -175,34 +189,35 @@ def add_listener(self, callback):
         self.listeners.append(callback)
 
     def listen_for_events(self, timeout_ms=30000):
-        """ Listen once for events. Use listen_forever to block indefinitely.
+        """Deprecated. sync now pulls events from the request.
+        This function just calls _sync()
 
         Args:
-            timeout (int): How long to poll the Home Server for before
+            timeout_ms (int): How long to poll the Home Server for before
                retrying.
         """
-        response = self.api.event_stream(self.end, timeout_ms)
-        self.end = response["end"]
-
-        for chunk in response["chunk"]:
-            for listener in self.listeners:
-                listener(chunk)
-            if "room_id" in chunk:
-                if chunk["room_id"] not in self.rooms:
-                    self._mkroom(chunk["room_id"])
-                self.rooms[chunk["room_id"]].events.append(chunk)
-                for listener in self.rooms[chunk["room_id"]].listeners:
-                    listener(chunk)
+        self._sync(timeout_ms)
 
     def listen_forever(self, timeout_ms=30000):
         """ Keep listening for events forever.
 
         Args:
-            timeout (int): How long to poll the Home Server for before
+            timeout_ms (int): How long to poll the Home Server for before
                retrying.
         """
         while(True):
-            self.listen_for_events(timeout_ms)
+            try:
+                self._sync(timeout_ms)
+            except MatrixRequestError as e:
+                print("A MatrixRequestError occured during sync.")
+                if e.code >= 500 or e.code == 404:
```

Comment:
Would we have already synced at this point? Generally apps die on 404, as it is taken to mean that the server has actually decommissioned the endpoint, rather than the a temporary failure